core/domain/stats_jobs_one_off.py

```diff
@@ -222,6 +226,22 @@ def reduce(exp_id, stringified_values):
                 state_session_ids_by_version[version][state_name].add(
                     session_id)
 
+                # Explicit logging of state names with + characters insted of
+                # spaces for future reference.
+                if '+' in state_name:
+                    state_name = state_name.replace('+', ' ')
+                    yield (
+                        'LO: State name %s of event (with ID %s created on '
+                        '%s) contains + instead of spaces.' % (
+                            state_name, value['event_id'],
+                            datetime.datetime.fromtimestamp(
+                                value['created_on']/1000)))
+
+                # There are a few state hit events which contain the pseudo end
+                # state as state name. These events are meant to be skipped.
+                if state_name == 'END':
```

Comment:
Any way you can tell for sure that it's pseudo? It's possible that some exps (after the pseudo-END state was removed) might have a legit state named with the string "END".

Reply:
@seanlip Oh yes. The pseudo END state should not be in exploration.states. Done.